AzureMonitor: Refresh documentation (#35371)
grafana/grafana: docs/sources/datasources/azuremonitor.md

| Default subscription    | (optional)                                                                                                                                                                                                                                                                                                              |
| Default workspace       | (optional)                                                                                                                                                                                                                                                                                                              |

## Azure Monitor query editor

The Azure Monitor data source has three different modes depending on which Azure service you wish to query:

- **Metrics** for [Azure Monitor Metrics](https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/data-platform-metrics)
- **Logs** for [Azure Monitor Logs](https://docs.microsoft.com/en-us/azure/azure-monitor/logs/data-platform-logs)
- Azure Resource Graph

### Querying Azure Monitor Metrics

> Azure Monitor Metrics collects numeric data from [supported resources](https://docs.microsoft.com/en-us/azure/azure-monitor/monitor-reference) and allows you to query them to investigate the health and utilisation of your resources to maximise availability and performance.

Author: No, i think that's correct.

Metrics are more lightweight than data in Azure Monitior Logs and only stores numeric data in a particular structure. Metrics is capable for supporting near real-time scenarios making it useful for fast detection of issues. Azure Monitor Logs can store a variety of different data types each with their own structure.

> > metrics screenshot

Author: Logs is mentioned to contrast it to Metrics, but you're right - I'll rework it.

#### Your first Azure Monitor Metrics query

1. Select the Metrics service
1. Select a resource to pull metrics from using the subscription, resource group, resource type, and resource fields.
1. Some resources, such as storage accounts, organise metrics under multiple metric namespaces. Grafana will pick a default namespace, but change this to see which other metrics are available.
1. Select a metric from the Metric field.

Optionally, you can apply further aggregations or filter by dimensions for further analysis.

1. Change the aggregation from the default average to show minimum, maximum or total values.
1. Set a specific custom time grain. By default Grafana will automatically select a time grain interval based on your selected time range.
1. For metrics that have multiple dimensions, you can split and filter further the returned metrics. For example, the Application Insights dependency calls metric supports returning multiple time series for successful vs unsuccessful calls.

> The options available will change depending on what is most relevent to the selected metric.

Reviewer: `relevent` -> `relevant`

| {{ dimensionvalue  }}       | _(Legacy for backwards compatibility)_ Replaced with the value of the first dimension       |

#### Supported Azure Monitor metrics

Not all metrics returned by the Azure Monitor Metrics API have values. To make it easier for you when building a query, the Grafana data source has a list of supported metrics and ignores metrics which will never have values. This list is updated regularly as new services and metrics are added to the Azure cloud. For more information about the list of metrics, refer to [current supported namespaces](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/supported_namespaces.ts).

### Querying Azure Monitor Logs

Azure Monitor Logs collects and organises log and performance data from [supported resources](https://docs.microsoft.com/en-us/azure/azure-monitor/monitor-reference) and makes many sources of data available to query together with the sophisticated [Kusto Query Language (KQL)](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/).

> Wheras Azure Monitor Metrics can only store numeric data in a simplified structure, Logs can store different data types each with their own strucutre and can perform complexe analysis of data using KQL.

Reviewer: Wheras?

# Azure Monitor data source

Grafana includes built-in support for Azure Monitor, the Azure service to maximise availablity and performance of your applications and services in the Azure Cloud. The Azure Monitor data source supports visualising data from three Azure services:

- **Azure Monitor Metrics** to collect numeric data from resources in your Azure account
- **Azure Monitor Logs** to collect log and performance data from your Azure account, and query using the powerful Kusto Language
- **Azure Resource Graph** to quickly query your Azure resources across subscriptions

This topic explains configuring, querying, and other options specific to the Azure Monitor data source. Refer to [Add a data source]({{< relref "add-a-data-source.md" >}}) for instructions on how to add a data source to Grafana.

## Azure Monitor settings

> To access Azure Monitor settings, hover your mouse over the **Configuration** (gear) icon, then click **Data Sources**, and then click the Azure Monitor data source. If you haven't already, you'll need to [add the Azure Monitor data source]({{< relref "add-a-data-source.md" >}}).
> 

Reviewer 2: ```suggestion
To access Azure Monitor settings, hover your mouse over the **Configuration** (gear) icon, click **Data Sources**, and then click the Azure Monitor data source. If you haven't already, you'll need to [add the Azure Monitor data source]({{< relref "add-a-data-source.md" >}}).

```

> You will need to create an app registration and service principal in Azure AD to authenticate the data source. See the [Azure documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in) for details on this. Alternatively, if you are hosting Grafana in Azure (e.g. App Service, or Azure Virtual Machines) you can configure the Azure Monitor data source to use Managed Identity to securely authenticate without entering credentials into Grafana. See [Configuring using Managed Identity](#todo) for more details.

Reviewer 2: ```suggestion
You must create an app registration and service principal in Azure AD to authenticate the data source. See the [Azure documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in) for configuration details. Alternatively, if you are hosting Grafana in Azure (e.g. App Service, or Azure Virtual Machines) you can configure the Azure Monitor data source to use Managed Identity to securely authenticate without entering credentials into Grafana. Refer to [Configuring using Managed Identity](#todo) for more details.
```

| Name                    | Description                                                                                                                                                                                                                                                                                                             |
| ----------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Authentication          | Enables Managed Identity. See [Configuring using Managed Identity](#todo) for more details.                                                                                                                                                                                                                             |

> | Azure Cloud             | The national cloud for your Azure account. For most users this will be the default "Azure". For more details, see [](https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud)                                                                                                     |

Reviewer 2: ```suggestion
| Azure Cloud             | The national cloud for your Azure account. For most users, this is the default "Azure". For more information, see [](https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud)                                                                                                     |
```
